Mage.Sets/src/mage/cards/a/AutomatedArtificer.java

```diff
@@ -75,23 +73,18 @@ class AutomatedArtificerManaCondition extends ManaCondition {
 
     @Override
     public boolean apply(Game game, Ability source) {
-        if (source == null) {
+        if (source == null || source.isActivated()) {
             return false;
         }
         switch (source.getAbilityType()) {
             case MANA:
             case ACTIVATED:
                 return true;
             case SPELL:
-                MageObject object = source.getSourceObject(game);
-                if (!(object instanceof StackObject) && !game.inCheckPlayableState()) {
```

Comment:
Why you remove inCheckPlayableStats, stackobject and commanderobject here?

Reply:
Because none of them seemed necessary, and all of the other ManaConditions for similar effects (spend only on spells of type X) didn't have them.

Reply:
Then make sure it works with commanders in command zone (allows to cast/pay for artifact commander and doesn't allow to cast/pay for non-artifact commander).

Reply:
I can confirm that the changed code allows paying for artifact commanders and does not allow paying for non-artifact commanders.